test/controllers/register/AreYouUKCompanyControllerSpec.scala

```diff
@@ -21,109 +21,86 @@ import data.SampleData._
 import forms.register.AreYouUKCompanyFormProvider
 import matchers.JsonMatchers
 import models.UserAnswers
-import org.mockito.ArgumentCaptor
+import play.api.inject.bind
 import org.mockito.ArgumentMatchers.any
 import org.mockito.Mockito.{times, verify, when}
 import org.scalatest.{OptionValues, TryValues}
 import org.scalatestplus.mockito.MockitoSugar
 import pages.register.AreYouUKCompanyPage
-import play.api.libs.json.{JsObject, Json}
+import play.api.libs.json.Json
 import play.api.mvc.Call
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 import play.twirl.api.Html
-import uk.gov.hmrc.viewmodels.{NunjucksSupport, Radios}
+import views.html.register.AreYouUkCompanyView
 
 import scala.concurrent.Future
 
-class AreYouUKCompanyControllerSpec extends ControllerSpecBase with MockitoSugar with NunjucksSupport with JsonMatchers with OptionValues with TryValues {
+class AreYouUKCompanyControllerSpec extends ControllerSpecBase with MockitoSugar with JsonMatchers with OptionValues with TryValues {
 
   private def onwardRoute = Call("GET", "/foo")
 
   private val formProvider = new AreYouUKCompanyFormProvider()
   private val form = formProvider()
 
-  private def areYouUKCompanyRoute = routes.AreYouUKCompanyController.onPageLoad().url
-  private def areYouUKCompanySubmitRoute = routes.AreYouUKCompanyController.onSubmit().url
-
+  private def areYouUKCompanyRoute: String = routes.AreYouUKCompanyController.onPageLoad().url
 
   private val answers: UserAnswers = userAnswersWithCompanyName.set(AreYouUKCompanyPage, true).success.value
 
   "AreYouUKCompany Controller" must {
 
     "return OK and the correct view for a GET" in {
-      when(mockRenderer.render(any(), any())(any())).thenReturn(Future.successful(Html("")))
+      val view = mock[AreYouUkCompanyView]
+
+      when(view.apply(any(), any(), any())(any(), any())).thenReturn(Html(""))
 
       val application = applicationBuilder(userAnswers = Some(userAnswersWithCompanyName))
         .overrides(
+          bind[AreYouUkCompanyView].toInstance(view)
         )
         .build()
-      val request = FakeRequest(GET, areYouUKCompanyRoute)
-      val templateCaptor = ArgumentCaptor.forClass(classOf[String])
-      val jsonCaptor = ArgumentCaptor.forClass(classOf[JsObject])
 
+      val request = FakeRequest(GET, areYouUKCompanyRoute)
       val result = route(application, request).value
 
       status(result) mustEqual OK
 
-      verify(mockRenderer, times(1)).render(templateCaptor.capture(), jsonCaptor.capture())(any())
-
-      val expectedJson = Json.obj(
-        "form"   -> form,
-        "submitUrl" -> areYouUKCompanySubmitRoute,
-        "radios" -> Radios.yesNo(form("value"))
-      )
-
-      templateCaptor.getValue mustEqual "register/areYouUKCompany.njk"
-      jsonCaptor.getValue must containJson(expectedJson)
+      verify(view, times(1)).apply(any(), any(), any())(any(), any())
```

Comment:
We don't need to mock the view just compare the result content to the injected view.
compareResultAndView(result, view)
Example in amend/* specs

Reply:
Same applies for the rest of the files.
We want to test that the view code is being executed properly.